src/driver/plugins.ts

```
@@ -1,20 +1,79 @@
+
         if (projection != null) {
             if (Object.keys(projection).length === 1 && projection['*']) {
+                // If schema has `select: true` for $vector or $vectorize, select: '*' will break with
+                // "wildcard ('*') only allowed as the only root-level path" error because Mongoose will
+                // add `$vector: 1` or `$vectorize: 1`. As a workaround, replace '*: 1' with including
+                // vector and vectorize.
+                if ($vector?.options?.select || $vectorize?.options?.select) {
+                    this.projection({ $vector: 1, $vectorize: 1 });
+                }
                 return;
             }
         }
-        const $vector = this.model.schema.paths['$vector'];
-        const $vectorize = this.model.schema.paths['$vectorize'];
-        if ($vector?.options?.select && (projection == null || !('$vector' in projection))) {
+        if ($vector?.options?.select && projectionDoesNotHaveProperty(projection, '$vector')) {
             this.projection({ ...projection, $vector: 1 });
         }
-        if ($vectorize?.options?.select && (projection == null || !('$vectorize' in projection))) {
+        if ($vectorize?.options?.select && projectionDoesNotHaveProperty(projection, '$vectorize')) {
             this.projection({ ...projection, $vectorize: 1 });
         }
     });
-}
\ No newline at end of file
+}
+
+function projectionDoesNotHaveProperty(projection: Record<string, unknown>, property: string) {
+    return projection == null || !(property in projection);
+}
+
+/**
+ * Mongoose plugin to validate arrays of numbers that have a `dimension` property. Ensure that the array
+ * is either nullish or has a length equal to the dimension.
+ *
+ * You do not need to call this function directly. Mongoose applies this plugin automatically when you call `setDriver()`.
+ */
+
+export function addVectorDimensionValidator(schema: Schema) {
```

Is this necessary? The Data API does the same thing on the server side anyhow, no? May as well let it throw the dimension error directly (like `astra-db-ts` does)

---

Not strictly necessary, more of a convention that Mongoose adds a custom validator for schema type options. Like `min`, `max`, `maxlength`, `minlength`, etc. So `await doc.validate()` can validate dimension mismatches.